pkg/ibmvpc/sg_analysis_test.go

```diff
@@ -107,3 +110,101 @@ func TestSGRule(t *testing.T) {
 	require.Equal(t, sgRule.index, 1)
 	require.Equal(t, ruleStr, "index: 1, direction: inbound,  conns: protocol: all, remote: 0.0.0.0/0, local: 10.240.10.0/32\n")
 }
+
+func TestNoLocalField(t *testing.T) {
+	cidr1t, _ := ipblock.FromCidrOrAddress("10.250.10.1")
+	cidr2t, _ := ipblock.FromCidrOrAddress("10.250.10.0/30")
+
+	c1 := connection.TCPorUDPConnection(netp.ProtocolString("UDP"), 5, 87, 10, 3245)
+	c2 := connection.TCPorUDPConnection(netp.ProtocolString("TCP"), 1, 100, 5, 1000)
+
+	rulesTest1 := []*SGRule{
+		{
+			target:      cidr1t,
+			connections: c1,
+			index:       1,
+			local:       ipblock.GetCidrAll(),
+		},
+		{
+			target:      cidr2t,
+			connections: c2,
+			index:       2,
+			local:       ipblock.GetCidrAll(),
+		},
+	}
+
+	target1, _ := ipblock.FromCidrOrAddress("10.250.10.0")
+	target2, _ := ipblock.FromCidrOrAddress("10.250.10.2")
+	target3, _ := ipblock.FromCidrOrAddress("10.250.10.1")
+
+	egressConnectivityMap := make(map[*ipblock.IPBlock]*ConnectivityResult)
+	mapAndAnalyzeSGRules(rulesTest1, false, egressConnectivityMap)
+	require.Equal(t, len(egressConnectivityMap), 1)
+	for _, CR := range egressConnectivityMap {
+		for remote, conn := range CR.allowedConns {
+			if target1.ContainedIn(remote) || target2.ContainedIn(remote) {
+				require.True(t, conn.Equal(c2))
+			}
+			if target3.ContainedIn(remote) {
+				require.True(t, c1.ContainedIn(conn) && c2.ContainedIn(conn))
+			}
+		}
+	}
```

Comment:
better be more explicit about expected content of  `CR.allowedConns` in the test, for example
can change to a switch over the set of expected remote `IPBlocks`, and for each case
verify that the connection is as expected. + also verify all switch cases were covered exactly once in the loop.
another option- assign an expected `connectivityResult` object , and use a method that compares equality of such objects. This can be used as table-driven tests for both tests TestNoLocalField and TestLocalField.